rule/package_directory_mismatch.go

```diff
@@ -0,0 +1,142 @@
+package rule
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/mgechev/revive/lint"
+)
+
+// PackageDirectoryMismatchRule detects when package name doesn't match directory name.
+type PackageDirectoryMismatchRule struct {
+	ignoredDirs *regexp.Regexp
+}
+
+const defaultIgnoredDirs = "testdata"
+
+// Configure the rule to exclude certain directories.
+func (r *PackageDirectoryMismatchRule) Configure(arguments lint.Arguments) error {
+	if len(arguments) < 1 {
+		return r.buildIgnoreRegex([]string{defaultIgnoredDirs})
+	}
+
+	args, ok := arguments[0].(map[string]any)
+	if !ok {
+		return fmt.Errorf("invalid argument type: expected map[string]any, got %T", arguments[0])
+	}
+
+	for k, v := range args {
+		if !isRuleOption(k, "ignore-directories") {
+			return fmt.Errorf("unknown argument %s for %s rule", k, r.Name())
+		}
+		ignoredDirs, ok := v.([]string)
+		if !ok {
+			return fmt.Errorf("invalid value %v for argument %s of rule %s, expected []string got %T", v, k, r.Name(), v)
+		}
+		return r.buildIgnoreRegex(ignoredDirs)
+	}
+
+	return nil
+}
+
+func (r *PackageDirectoryMismatchRule) buildIgnoreRegex(ignoredDirs []string) error {
+	if len(ignoredDirs) == 0 {
+		return nil
+	}
+
+	patterns := make([]string, len(ignoredDirs))
+	for i, dir := range ignoredDirs {
+		patterns[i] = regexp.QuoteMeta(dir)
+	}
+	pattern := "(" + strings.Join(patterns, "|") + ")"
```

Comment:
I think parenthesis are not needed here
```suggestion
	pattern := strings.Join(patterns, "|") 
```